Source/Diagnostics/RecordingPlaneDiagnostics.H

```diff
@@ -0,0 +1,84 @@
+#ifndef WARPX_RECORDINGPLANEDIAGNOSTICS_H
+#define WARPX_RECORDINGPLANEDIAGNOSTICS_H
+
+#include "Diagnostics.H"
+#include "Diagnostics/ComputeDiagFunctors/ComputeDiagFunctor.H"
+#include "Utils/Parser/IntervalsParser.H"
+
+#include <string>
+
+
+class RecordingPlaneDiagnostics final
+    : public Diagnostics
+{
+public:
+    RecordingPlaneDiagnostics (int i, std::string name);
+private:
+    /** Read user-requested parameters for station diagnostics */
+    utils::parser::SliceParser m_intervals;
+    void ReadParameters ();
+
+    /** Station location where field values are recorded in time */
+    amrex::Real m_station_loc;
+
+    /** The direction normal to the station slice where fields are recorded in time*/
+#if defined(WARPX_DIM_1D_Z)
+    enum struct StationNormalDir {z=0};
+#elif defined(WARPX_DIM_XZ)
+    enum struct StationNormalDir {x=0, z};
+#elif defined(WARPX_DIM_RZ)
+    enum struct StationNormalDir {r=0, z};
+#else
+    enum struct StationNormalDir {x=0, y, z};
+#endif
+    StationNormalDir m_station_normal = StationNormalDir::z;
+
+    /** whether to flush at this time step
+     * \param[in] step current time step
+     * \param[in] i_buffer index of a back-transformed snapshot
+     * \param[in] force_flush if true, return true for any step
+     * \return bool, whether to flush
+     */
+    bool DoDump (int step, int i_buffer, bool force_flush=false) override;
```

Comment:
A comment not specific to this PR. A lot of virtual functions in Diagnostics have default parameters. This is probably a bad idea. We might consider change it. @ax3l Maybe we should discuss it in the next meeting.

 https://google.github.io/styleguide/cppguide.html#Default_Arguments